Contributions/Kernel_Self_Monitoring.md

```diff
@@ -0,0 +1,117 @@
+# **Kernel Self Monitoring**
+
+## Index
+[References](#References)
+
+[Purpose of the Document](#Purpose-of-the-Document)
+
+[Structure of the Document](#Structure-of-the-Document)
+
+[Introduction - Linux Kernel: practical safety problems](#Introduction---Linux-Kernel-practical-safety-problems)
+
+[The concept - Safety-oriented self monitoring](#The-concept---Safety-oriented-self-monitoring)
+
+[Constraints for a safety-oriented kernel self monitor](#Constraints-for-a-safety-oriented-kernel-self-monitor)
+
+[Targets for self monitoring activity](#Targets-for-self-monitoring-activity)
+
+
+## **References**
+
+## **Purpose of the Document**
+These are guidelines for safety-hardening certain aspects of the Linux kernel, through self monitoring.
+
+They can act as a touchstone against candidate designs, to avoid known pitfalls. 
+
+## **Structure of the document**
+- The document presents a brief overview of what safety problems can be found while using Linux, and how they can be mitigated through self-monitoring.
+- The next sections, instead, discuss more in detail typical pitfalls that must be avoided, to make the monitoring useful and what goals should be targeted.
+
+## **Introduction - Linux Kernel: practical safety problems**
+The Linux Kernel has many desirable features that make it a prime candidate for many use cases.
+
+However, its lacks of both an intrinsic safety design and safety-oriented features make it challenging to use it as-it-is, for safety applications.
+
+Therefore, there is a risk of having a Linux-based system to experience undetected internal interference, which might compromise the set of safety goals the system must support.
+
+It is a customary mitigation to introduce system-level monitoring, in various forms.
+
+To mention a couple of examples:
+- Watchdogs, to detect abnormal delays either in processing events or performing actions.
+- Output vetting, to detect abnormal actuator control signals produced by the system.
+
+These external safety measures, though, are not always sufficient, because they might have limited ability to observe the internal status of the kernel, or it might be desirable to exert a tighter control over the evolution of its internal states.
```

Comment:
In the statement above I would add "especially in scenarios where the Kernel is used to enable and manage complex safety workloads"